### PR TITLE
Add support for custom webpack config

### PR DESCRIPTION
**Pull request by connor-baer**
Based on #112. 

## Changes

1. Add a `--webpack` option to the CLI to point to a custom config
2. Automatically look for a `webpack.config.js` from the CLI using `find-up` as [done here in x0](https://github.com/c8r/x0/blob/master/cli.js#L114-L119).
3. Use `webpack-merge` to merge with the built-in config

### CODE REVIEW

**Comment by connor-baer** [comment]
I'm on the road and my laptop is running out of battery, will test and finish this later.

**Comment by connor-baer** [comment]
@jxnblk This is ready for review. 

I wasn't quite sure how to make sure `mdx-deck/loader` is still used. The original ticket asks for a way to add additional remark mdPlugins, so those would have to be merged in somehow. Maybe a note in the documentation would suffice?

**Comment by jxnblk** [comment]
Thanks! This looks great!

> I wasn't quite sure how to make sure mdx-deck/loader is still used.

Yeah, maybe that's something to note in the docs, but I'm sure people will break the setup. Could be fine just mentioning this in the docs.

File: lib/config.js [hunk 1]
```diff
@@ -107,7 +108,7 @@ const baseConfig = {
 }
 
 const createConfig = (opts = {}) => {
-  const config = Object.assign({}, baseConfig)
+  const config = merge(baseConfig, opts.webpack)
```
Comment (jxnblk) [inline lib/config.js]: I can't remember, but do you know if this clones the `baseConfig`?

Comment (connor-baer) [reply lib/config.js]: Looks like it: https://github.com/survivejs/webpack-merge/blob/master/src/index.js#L15

**Comment by connor-baer** [comment]
I've added a note about the default `mdx-deck/loader`.

**Comment by jxnblk** [comment]
Looks great, thanks! 🎉

**Pull request merged**